superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -373,27 +374,20 @@ export function dbReducer(
         },
       };
     case ActionType.setSSHTunnelLoginMethod:
-      if (action.payload.login_method === AuthType.privateKey) {
-        const { password, ...rest } = trimmedState?.ssh_tunnel ?? {};
-        return {
-          ...trimmedState,
-          ssh_tunnel: {
-            ...rest,
-          },
-        };
-      }
-      if (action.payload.login_method === AuthType.password) {
-        const { private_key, private_key_password, ...rest } =
-          trimmedState?.ssh_tunnel ?? {};
-        return {
-          ...trimmedState,
-          ssh_tunnel: {
-            ...rest,
-          },
-        };
+      let ssh_tunnel;
+      if (trimmedState?.ssh_tunnel) {
+        // remove any attributes that are considered sensitive
+        ssh_tunnel = pick(trimmedState.ssh_tunnel, [
```

Comment:
Your tunnel has no credentials in it if you just `pick` these.

Reply:
got it!